repository/src/main/java/org/apache/atlas/repository/store/graph/v2/IndexRepairService.java

```diff
@@ -0,0 +1,234 @@
+            AtlasVertex vertex = vertices.next();
+            // Check if vertex is corrupted
+            if (CollectionUtils.isEmpty(vertex.getPropertyKeys())){
+                return vertex.getId() instanceof Long ? Optional.of((Long) vertex.getId()) : Optional.empty();
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Repair single index for a given qualifiedName
+     */
+    public RepairResult repairSingleIndex(String qualifiedName) throws AtlasBaseException {
+        LOG.info("Starting single index repair for QN: {}", qualifiedName);
+
+        Optional<Long> corruptedVertexId = findCorruptedSingleVertex(qualifiedName);
+
+        if (!corruptedVertexId.isPresent()) {
+            LOG.info("No corrupted vertex found for QN: {}", qualifiedName);
+            return new RepairResult(false, null, "No corruption detected in single index");
+        }
+
+        Long vertexId = corruptedVertexId.get();
+
+        // For single index, we can rename the property to remove it from index
+        try {
+            AtlasVertex vertex = graph.getVertex(vertexId.toString());
+            if (vertex != null) {
+                // Rename the indexed property to remove from index
+                String tempPropertyName = Constants.UNIQUE_QUALIFIED_NAME + "_corrupted_" + System.currentTimeMillis();
+                Object oldValue = vertex.getProperty(Constants.UNIQUE_QUALIFIED_NAME, String.class);
+
+                if (oldValue != null) {
+                    vertex.setProperty(tempPropertyName, oldValue);
+                    vertex.removeProperty(Constants.UNIQUE_QUALIFIED_NAME);
+                }
+
+                // Now remove the vertex
+                graph.removeVertex(vertex);
+                graph.commit();
+
+                LOG.info("Successfully repaired single index for QN: {}, VertexId: {}",
+                        qualifiedName, vertexId);
+
+                return new RepairResult(true, vertexId, "Successfully repaired single index");
+            }
+        } catch (Exception e) {
+            LOG.error("Failed to repair single index for vertex: {}", vertexId, e);
+            throw new AtlasBaseException(AtlasErrorCode.INTERNAL_ERROR,
+                    "Failed to repair single index: " + e.getMessage());
+        }
+
+        return new RepairResult(false, null, "Vertex not found");
+    }
+
+    /**
+     * Find corrupted vertex using composite index
+     */
+    public Optional<Long> findCorruptedCompositeVertex(String qualifiedName, String typeName) {
+        if (StringUtils.isBlank(qualifiedName) || StringUtils.isBlank(typeName)) {
+            return Optional.empty();
+        }
+
+        AtlasGraphQuery query = graph.query()
+                .has(Constants.UNIQUE_QUALIFIED_NAME, qualifiedName)
+                .has(Constants.TYPE_NAME_PROPERTY_KEY , typeName);
+
+        Iterator<AtlasVertex> vertices = query.vertices().iterator();
+
+        if (vertices.hasNext()) {
```

Comment:
Realised one thing: Please adda a failsafe if asset is not corrupted we will throw error. If important attrs are present we can't update. So we don't accidentally remove index record for an asset.